src/Webapi/Webapi__TextDecoder.res

```diff
@@ -0,0 +1,35 @@
+type t = {
+  encoding: string,
+  fatal: bool,
+  ignoreBOM: bool,
+}
+
+type decoderOptions
+
+type decodeOptions
+
+%%private(
+  @new external _makeWithOptions: (string, decoderOptions) => t = "TextDecoder"
+
+  @obj
+  external makeDecoderOptions: (
+    ~fatal: option<bool>=?,
+    ~ignoreBOM: option<bool>=?,
+    unit,
+  ) => decoderOptions = ""
+)
+
+@new external make: unit => t = "TextDecoder"
+
+let makeWithOptions = (~encoding="utf-8", ~fatal=?, ~ignoreBOM=?, ()) =>
+  _makeWithOptions(encoding, makeDecoderOptions(~fatal, ~ignoreBOM, ()))
```

Comment:
While I appreciate the sentiment to make the bindings more usable, we want these bindings to be zero-cost as much as possible. In this case ReScript 10 should have a new feature that makes `makeDecoderOptions` unnecessary, but it will be easier to switch to the upcoming style if the exported API takes a `decoderOptions`.

Please remove the private wrappers and export the more difficult to use way for now; here and below.
```suggestion
@obj
external makeDecoderOptions: (
  ~fatal: option<bool>=?,
  ~ignoreBOM: option<bool>=?,
  unit,
) => decoderOptions = ""

@new external make: unit => t = "TextDecoder"
@new external makeWithOptions: (string, decoderOptions) => t = "TextDecoder"
```